src/app/modules/zxing-scanner/browser-code-reader.ts

```diff
@@ -159,20 +159,24 @@ export class BrowserCodeReader {
 
         const imageCapture = new ImageCapture(this.track);
         const photoCapabilities = imageCapture.getPhotoCapabilities().then((capabilities) => {
-            this.torchCompatible.next(!!capabilities.torch);
-            compatible = capabilities.torch;
+            compatible = (!!capabilities.torch) ||
+                ('fillLightMode' in capabilities && capabilities.fillLightMode.length !== 0);
+            this.torchCompatible.next(compatible);
         });
         return compatible;
     }
 
-    public setTorch(on: boolean): void {
-        this.torchCompatible.subscribe(compatible => {
-            if (compatible) {
+    public setTorch(on: boolean) {
+        if (this.torchCompatible.value) {
+            if (on) {
                 this.track.applyConstraints({
-                    advanced: [<any>{ torch: on }]
+                    advanced: [<any>{ torch: true }]
                 });
+            } else {
+                // TODO: Restarting stream is needed as torch false is not working on chrome.
```

Comment:
Can we fix this TODO before merge?

Reply:
Can I store the parameters given to the `decodeFromInputVideoDevice`? That way I could restart the device with the video.

Reply:
I think the code reader should know from where the decode is happening. Seems like we could use `startDecodeFromStream` for a restart, but looks like the `callbackFn` is still missing; looks like the only way to do a reset is by storing this little guy too. @werthdavid what you think?